include/mbedtls/config_adjust_legacy_crypto.h

```diff
@@ -135,6 +135,47 @@
 #define MBEDTLS_PK_PARSE_C
 #endif
 
+/* Helpers to state that each key is supported either on the builtin or PSA side. */
```

Comment:
Looking again, I'm questioning whether this is the right place to define `MBEDTLS_ECP_HAVE_xxx`. These symbols are similar to `MBEDTLS_MD_CAN_xxx` which are defined in `md.h` and `MBEDTLS_PK_CAN_xxx` which are defined in this file.

Normally code that depends on `MBEDTLS_MD_HAVE_xxx` would have to include `md.h` since it would be calling `mbedtls_md_xxx` functions. However, there could be indirect dependencies where a module A calls another module B that calls md, and A itself doesn't call md functions but needs to have e.g. a buffer size calculation depend on which md algorithms are supported. So there's a risk that we forget to `#include "mbedtls/md.h"` from A and the buffer size is wrong.

Thus I think that we should move the definitions of `MBEDTLS_MD_HAVE_xxx` here.

This is out of scope of this pull request, which is already getting very long. So please don't do it here, let's remember to do it in a follow-up.

Reply:
Created https://github.com/Mbed-TLS/mbedtls/issues/8257 to track this, and added in to the "Driver-only cleanups" EPIC (which due to this issue is no longer just about ECC). As a result, marking this converstaas resolved, please shout if you disagree.